plugins/woocommerce/includes/class-wc-structured-data.php

```diff
@@ -577,4 +577,36 @@ public function generate_order_data( $order, $sent_to_admin = false, $plain_text
 
 		$this->set_data( apply_filters( 'woocommerce_structured_data_order', $markup, $sent_to_admin, $order ), true );
 	}
+
+	/**
+	 * Check if a GTIN is valid.
+	 * A GTIN is valid if:
+	 * Is a positive numeric string with a length of 8,12,13 or 14.
+	 *
+	 * @see https://schema.org/gtin
+	 * @param string $gtin The GTIN to check.
+	 * @return bool True if valid. False otherwise.
+	 */
+	public function is_valid_gtin( $gtin ) {
+		if ( $gtin && is_numeric( $gtin ) && $gtin > 0 ) {
```

Comment:
> Just curious how useful the $gtin > 0 part is here. I'm assuming we are checking it's positive, but because we are checking a string of just numeric digits, it doesn't seem like it would do anything.

if we remove `$gtin > 0` then -1000000 will be a valid GTIN which is not correct.

Reply:
> For example if we call it without using prepare_gtin than the following would return true (and is a value we are allowed to save in the input field):

`is_valid_gtin( '12-345-6' )` is not valid. See https://github.com/woocommerce/woocommerce/pull/50926/files/ec9cd9841dfd9fc437097892531bfdc6c0ca5836#diff-d28b2de048c5c8d3906ce3c0ee57e836e618817e229d422391f5f0d23109885dR41

Reply:
Thanks for the clarification. At the time I was thinking the "prepared" GTIN was needed to strip out all dashes. But the `is_numeric` + `> 0` takes care of invalidating those if it's left in.

But what I was trying to relay is that isn't strict enough, as `is_numeric` is very lenient, for example `123.4e-5` is numeric, larger than 0 and has a string length of 8. Also while we prevent numbers smaller than 0 to prevent a negative symbol we don't do anything about a positive symbol `+1234567` passes all those checks.

Instead of making the checks really strict to catch all cases, I wonder if it isn't just simpler to call `prepare_gtin` from within `is_valid_gtin` so we only need to take "prepared" numbers into consideration.

Reply:
I would advocate for having two separated functions to separate concerns. One for validating and one for preparing let's say. Otherwise, as you mentioned, since it's a public function someone can do something like this:

```php
 $gtin = +1234567;
 $is_valid = validate_gtin( $gtin ); // return true (because is calling prepare gtin inside); 
 if ( $is_valid ) {
     echo $gtin;
 }
```

So I would advocate for tweaking the `validate_gtin` function even more to avoid these edge cases like floats or positive symbols you caught. Or having a new public `get_formatted_gtin` that calls `prepare_gtin` + `validate_gtin` and makes those private. So we can just test `get_formatted_gtin`

WDYT?

Reply:
I tweaked for now `is_valid_gtin` --> https://github.com/woocommerce/woocommerce/pull/50926/commits/9f8f75357e587c6c2828bd9fe41a26bacd161afc